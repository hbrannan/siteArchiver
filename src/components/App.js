import React from 'react'
import SearchDisplay from './SearchDisplay'
import SiteDisplay from '../containers/SiteDisplay'
import PropTypes from 'prop-types'
// import './App.less';

const App = ({isDisplayingSite}) => {
  if (!isDisplayingSite) return <SearchDisplay />;
  else return <SiteDisplay />;
}

App.propTypes = {
  isDisplayingSite: PropTypes.bool.isRequired
};

export default App;
