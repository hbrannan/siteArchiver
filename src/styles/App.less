//External Styles
@import './_vendor_normalize.less';

//Globally defined styles
@import './_defaults.less';
@import './_utils.less';
@import './_variables.less';

//Components (from general -> specific use cases)
@import 'components/_Form.less';
@import 'components/_TopFiveList.less';
@import 'components/_SiteDisplay.less';

.display__container {
  margin: .75rem;
  .flex-display(flex);
  .flex-direction(row);
  .flex-wrap(wrap);
  .justify-content(center);
}

.display__header {
  width: 100%;
  .flex-display(flex);
  .flex-direction(row);

  h1 {
    width: 100%;
    text-align: center;
  }
}

