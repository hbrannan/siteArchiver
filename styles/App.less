//External Styles
@import './styles-global/normalize.less';

//Globally defined styles
@import './styles-global/defaults.less';
@import './styles-global/variables.less';
@import './styles-global/utils.less';

//Components
@import './components/_Form.less'
@import './components/_TopFiveList.less'
