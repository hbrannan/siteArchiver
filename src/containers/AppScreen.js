import { connect } from 'react-redux'
import App from '../components/App'

const mapStateToProps = state => ({
  isDisplayingSite: state.app.isDisplayingSite
});

const AppScreen = connect(
  mapStateToProps
)(App);

export default AppScreen
