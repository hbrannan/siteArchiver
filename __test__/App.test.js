import React from 'react'
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { shallow, mount } from 'enzyme'
import App from '../src/components/App'
import AppScreen from '../src/containers/AppScreen'
import state from './state-stub'

const middlewares = [];
const mockStore = configureMockStore(middlewares)({app: {isDisplayingSite: false}});
jest.mock('../src/containers/SiteDisplay', () => 'div className="site__container"')
jest.mock('../src/components/SearchDisplay', () => 'div className="search__container"')

describe('App Screen', () => {
  it('should render', () => {
    const wrapper = shallow(
      <Provider store={mockStore}>
        <AppScreen />
      </Provider>
    );
    expect(wrapper.find(AppScreen).length).toEqual(1);
  });

});

describe('App (Snapshot)', () => {
  it('consistently renders the App shell', () => {
    const component = renderer.create(
      <Provider store={mockStore}>
        <AppScreen />
      </Provider>).toJSON();

    expect(component).toMatchSnapshot();
  });
});


/*

Todo:

  refactor: to use moduleNameMapper option and a stub file in Jest config
  (rather than current weback.test config & test env & plugin)
*  *  *    https://github.com/facebook/jest/issues/870
  Jest config:

  "moduleNameMapper": {
    "\.scss$": "path/to/SCSSStub.js"
  }
  SCSSStub.js:

  module.exports = {}; // Or other stub data for all SCSS modules.

*  *  *
  Resources
  - Understanding a componentâ€™s contract: defines the expected behavior of your component and
    what assumptions are reasonable to have about its usage
      - What it renders  - props  - state held  - response to user interactions
      - context rendered in  - side effects as a part of component lifecycle
      - performance of methods called on instance  -

  Test shouldn't need to exactly duplicate app.
  Test shouldn't duplicate library code.
  Trivial or important detail? Does this aspect affect the component's public API?

  https://www.codementor.io/vijayst/unit-testing-react-components-jest-or-enzyme-du1087lh8
  https://medium.freecodecamp.org/the-right-way-to-test-react-components-548a4736ab22
  https://github.com/reactjs/redux/blob/master/docs/recipes/WritingTests.md
*/
