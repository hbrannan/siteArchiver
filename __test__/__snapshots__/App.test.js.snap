// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App (Snapshot) consistently renders the App shell 1`] = `
<div>
  <h1>
    Web Crawler
  </h1>
  <div
    className="form__container"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Search for a site"
      />
    </form>
    <div
      className="form__response"
    />
  </div>
  <div />
</div>
`;
